packages/cursorless-engine/src/processTargets/modifiers/scopeHandlers/SentenceScopeHandler/SentenceSegmenter.ts

```diff
@@ -0,0 +1,60 @@
+import * as sbd from "sbd";
+import { MatchedText } from "../../../../util/regex";
+
+// A sentence starts with a letter
+const leadingOffsetRegex = /\p{L}/u;
+// A line with no letters is invalid and breaks sentences
+const invalidLineRegex = /(\n[^\p{L}]*\n)/gu;
+
+const options: sbd.Options = {
+  ["newline_boundaries"]: false,
+  ["preserve_whitespace"]: true,
+};
+
+export default class SentenceSegmenter {
+  *segment(text: string): Iterable<MatchedText> {
+    const sentences = sbd.sentences(text, options);
+    let index = 0;
+
+    for (const sentence of sentences) {
+      const parts = sentence.split(invalidLineRegex);
+
+      for (const part of parts) {
+        if (!isInvalidLine(part)) {
```

Comment:
if we're using invalidLineRegex `invalidLineRegex` to split, then I don't think we'd get any invalid lines, right?

Reply:
The regex has a capture group. If you split on a regex with the capture group that content will be in the final result. We need that since we need to calculate a correct index for the start of each sentence.